testing/citest/tests/appengine_smoke_test.py

```diff
@@ -0,0 +1,339 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Integration test for App Engine. Primarily tests the deploy operation and upsert load balancer pipeline stage,
+which are relatively complex and not well covered by unit tests.
+
+Sample Usage:
+    Assuming you have created $PASSPHRASE_FILE (which you should chmod 400)
+    and $CITEST_ROOT points to the root directory of the citest library. The passphrase file
+    can be omited if you run ssh-agent and add .ssh/compute_google_engine. 
+
+  PYTHONPATH=$CITEST_ROOT \
+    python spinnaker/testing/citest/tests/appengine_smoke_test.py \
+    --gce_ssh_passphrase_file=$PASSPHRASE_FILE \
+    --gce_project=$PROJECT \
+    --gce_zone=$ZONE \
+    --gce_instance=$INSTANCE 
+or
+  PYTHONPATH=$CITEST_ROOT \
+    python spinnaker/testing/citest/tests/appengine_smoke_test.py \
+    --native_hostname=host-running-smoke-test
+"""
+
+import sys
+
+import citest.gcp_testing as gcp
+import citest.json_contract as jc
+import citest.json_predicate as jp
+import citest.service_testing as st
+
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+import spinnaker_testing.frigga as frigga
+import citest.base
+
+
+class AppengineSmokeTestScenario(sk.SpinnakerTestScenario):
+  """Defines the scenario for the integration test.
+  
+  We're going to:
+    Create a Spinnaker Application
+    Create a Spinnaker Server Group (implicitly creates a Spinnaker Load Balancer)
+    Create a Pipeline with the following stages
+      - Deploy
+      - Upsert Load Balancer
+    Delete Load Balancer (implicitly destroys the Spinnaker Server Groups created within this test)
+    Delete Application
+  """
+  @classmethod
+  def new_agent(cls, bindings):
+    return gate.new_agent(bindings)
+
+  @classmethod
+  def initArgumentParser(cls, parser, defaults=None):
+    """Initialize command line argument parser."""
+    super(AppengineSmokeTestScenario, cls).initArgumentParser(
+          parser, defaults=defaults)
+    parser.add_argument('--source_repo_url', default=None, 
+                        help='URL of a source code repository used by Spinnaker to deploy to App Engine.')
```

Comment:
Yes - this argument and the one you've pointed out below are required. The test now throws an error if you haven't supplied these.